dashboard/models.py

```diff
@@ -150,13 +151,46 @@ def get_supported_courses(self):
             logger.info("Courses did not exist", exc_info = True)
         return []
 
+    def earliest_start_datetime(self) -> Optional[datetime]:
+        """Get the earliest start date of courses in the QuerySet
 
-class CourseManager(models.Manager):
-    def get_queryset(self):
-        return CourseQuerySet(self.model, using=self._db)
+        :return: Earliest start date of courses in the QuerySet, or None if no course dates found
+        :rtype: datetime
+        """
+        sorted_courses = sorted(self.all(), key=lambda course: course.course_date_range.start)
 
-    def get_supported_courses(self):
-        return self.get_queryset().get_supported_courses()
+        earliest_start = None
+        if len(sorted_courses) > 0:
+            earliest_course = sorted_courses[0]
+            earliest_start = earliest_course.course_date_range.start
+            logger.info(f"Earliest start datetime for all courses: {earliest_start.isoformat()} found in course {earliest_course.canvas_id}")
+        else:
+            logger.info(f"No course listed. Returning None as the earliest_start_datetime_of_course. ")
+        return earliest_start
+
+    def update_all_data_last_updated(self, updated_date=None):
+        """ Updates the last cron run to now for all courses in QuerySet
+        """
+        # Finally update the time on all courses updated
+        if updated_date is None:
+            updated_date = datetime.now(pytz.UTC)
```

Comment:
Nit-pick: What's the reason for making `updated_date` optional? I don't think you're doing this now anywhere, and even if you wanted to, you could just pass `datetime.now(tz=pytz.UTC)` to this.

You could add a type hint here for `updated_date` and the return type (`None`).

Reply:
Now that I'm looking at this more, would it be so bad to remove this entirely and just use `SomeQuerySet.update(data_last_updated=some_datetime)`?

Reply:
Since I'm only calling it one it seems fine. I was reading about how explicitly named methods can be better than more complex ones on an object, but this one is pretty straightforward.